core/deployment/src/main/java/io/quarkus/deployment/dev/QuarkusDevModeLauncher.java

```diff
@@ -376,14 +376,30 @@ protected void prepare() throws Exception {
         if (debug != null && debug.equalsIgnoreCase("client")) {
             args.add("-agentlib:jdwp=transport=dt_socket,address=" + debugHost + ":" + port + ",server=n,suspend=" + suspend);
         } else if (debug == null || !debug.equalsIgnoreCase("false")) {
-            // make sure the debug port is not used, we don't want to just fail if something else is using it
-            // we don't check this on restarts, as the previous process is still running
+            // if the debug port is used, we want to make an effort to pick another one
```

Comment:
shouldn't we print a info/warn that is locating a free port because specified one is not working?

btw. do we really need the 5 retries?

Reply:
Good point about the error message - added. 

> btw. do we really need the 5 retries?

It's just a detail honestly...

Reply:
just tested - and yes; we really need to print something about it not using the expected configured port.